### DISCUSSION TITLE
Object Namings

### DISCUSSION THREAD

**Discussion by Stwissel** [beyonddemise/vertx-couchdb-client#25] [Polls]
We have a lot of classes prefixed with `Couch...`

Secondary: Do we need the prefix for all Classes or just the main ones

**Comment by Stwissel** [comment]
@rjtmahinay @jmgang

**Reply by rjtmahinay** [reply]
@Stwissel Here's my thought

At first, I wondered if we could omit the DB or Db part since it was understandable that CouchClient is referring to CouchDB but there's Couchbase so others may think we're supporting both NoSQL databases.

On my end, I don't see any strong reason why we don't want to use CouchDb as the prefix.  Most developers follow camelCase and will treat DB as Db.

**Reply by Stwissel** [reply]
The argument against the capital D would be: CamelCase is used to separate logical parts. `Couchdb` would be a prefix before `Client`, `Database` or `Document`. The purpose is to clearly distinguish e.g. `CouchdbDatabase` from `PostgresDatabase`. The capital `D` creates the impression there is `Couch` and there is `Db`, which actually never get separated. At least not in our code base.
